spacesuit/src/value.rs

```diff
@@ -126,10 +126,24 @@ impl Into<Scalar> for SignedInteger {
 }
 
 impl Add for SignedInteger {
-    type Output = SignedInteger;
+    type Output = Option<SignedInteger>;
 
-    fn add(self, rhs: SignedInteger) -> SignedInteger {
-        SignedInteger(self.0 + rhs.0)
+    fn add(self, rhs: SignedInteger) -> Option<SignedInteger> {
+        match self.0.checked_add(rhs.0) {
```

Comment:
`checked_add` here is "checked" w.r.t. i128 overflows, not our funky 65-bit integer. Since we know that our i128 are guaranteed to be in 64-bit range by absolute value, they will never overflow 128 bits, so we can:

1. simply add them together
2. check that the result is ≤ than `(u64::MAX as i128)` and ≥ than `-(u64::MAX as i128)`.